ros/src/tl_detector/light_classification/ssd_tl_classifier.py

```diff
@@ -34,37 +47,112 @@ def load_graph(graph_file):
+        thickness = (pil_image.size[0] + pil_image.size[1]) // 300
+
+        draw = ImageDraw.Draw(pil_image)
+
+        for i, c in enumerate(classes):
+            score = scores[i]
+            predicted_class = self.labels_dict[c]
+            box = box_coords[i]
+
+            label = '{} {:.2f}'.format(predicted_class, score)
+            label_size = draw.textsize(label, font)
+
+            top, left, bottom, right = box
+            top = max(0, np.floor(top + 0.5).astype('int32'))
+            left = max(0, np.floor(left + 0.5).astype('int32'))
+            bottom = min(pil_image.size[1], np.floor(bottom + 0.5).astype('int32'))
+            right = min(pil_image.size[0], np.floor(right + 0.5).astype('int32'))
+
+            if top - label_size[1] >= 0:
+                text_origin = np.array([left, top - label_size[1]])
+            else:
+                text_origin = np.array([left, top + 1])
+
+            for j in range(thickness):
+                draw.rectangle([left + j, top + j, right - j, bottom - j], outline=self.labels_dict[c])
+
+            draw.rectangle([tuple(text_origin), tuple(text_origin + label_size)], fill=self.labels_dict[c])
+
+            draw.text(text_origin, label, fill=(0, 0, 0), font=font)
+        return np.asarray(pil_image)
+
     def _classify(self, image):
         image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
-        image_np = np.expand_dims(np.asarray(image, dtype=np.uint8), 0)
+        image_resized = cv2.resize(image, (300, 300))
+        image_np = np.expand_dims(np.asarray(image_resized, dtype=np.uint8), 0)
+
         # Actual detection
-        (boxes, scores, classes) = self.sess.run([self.detection_boxes, self.detection_scores, self.detection_classes],
-                                                 feed_dict={self.image_tensor: image_np})
+        (boxes, scores, classes) = self.sess.run([self.detection_boxes, self.detection_scores,
+                                                  self.detection_classes], feed_dict={self.image_tensor: image_np})
 
         # Remove unnecessary dimensions
+        boxes = np.squeeze(boxes)
         scores = np.squeeze(scores)
         classes = np.squeeze(classes)
 
-        for i, clazz in enumerate(classes):
-            rospy.logdebug('class = %s, score = %s', self.labels_dict[classes[i]], str(scores[i]))
-            # if red or yellow light with score more than confidence threshold
-            if (clazz == 2) and scores[i] > self.confidence:
-                return TrafficLight.RED, None
-            if (clazz == 3) and scores[i] > self.confidence:
-                return TrafficLight.YELLOW, None
-
-        return TrafficLight.UNKNOWN, None
+        boxes, scores, classes = self.filter_boxes(boxes, scores, classes)
+
+        for i, c in enumerate(classes):
+            rospy.logdebug('class = %s, score = %s', self.labels_dict[c], str(scores[i]))
+
+        if len(classes) == 0:
```

Comment:
Automatic code reviewer CodeFactor said that writing `if not classes: ...` is more pythonic in this case.

Reply:
`if not classes:` doesn't work with numpy, it raises "ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()"......used `if classes.size == 0` instead.